Isabella Montenegro Cadena :)

**Caso de estudio**
 En este documento se desarrolla y analiza un modelo Naive Bayes para una entidad financiera que busca fortalecer su sistema de prevención de lavado de activos (SARLAFT) utilizando analítica de datos.  De acuerdo con lo anterior las variables son:

 Edad: es la edad que posee el empleado.

 Ingresos: Ingresos recibidos por una personas como resultado de una actividad comercial.

 Egresos: Son los gastos mensuales de una persona, Gastos de una personaa cargo, hijos y servicios públicos.

Numero Tarjetas de Credito: Es el número de tarjetas de crédito activas asociadas a la persona, que refleja su acceso a instrumentos financieros.

Monto Transado de Tarjetas (USD): Valor total de las transacciones realizadas con tarjetas de crédito, en dólares estadounidenses, indicador del volumen real de movimiento financiero.

Porcentaje Crecimiento Patrimonio: Variación porcentual del patrimonio de la persona en un período determinado, utilizada para detectar incrementos patrimoniales no explicados.


 Lavado de Activos: Es el proceso ilegal para ocultar el movimiento de dinero o bienes de una organizacion o persona.




0. Cargamos las librerias del trabajo.

In [ ]:
import numpy as np
import pandas as pd

from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import confusion_matrix

1. Cargamos la Base de Datos.

In [ ]:
nxl = "/content/2. LavadoActivos.xlsx"
XBD = pd.read_excel(nxl)
XBD.head()

XD=XBD[["edad","ingresos_usd", "gastos_usd", "num_tarjetas_credito","monto_transado_tarjetas_usd","porcentaje_crecimiento_patrimonio"]] #Variables de trabajo
XD.head(10)
yd=XBD[["lavado_activos"]]
yd.head()

FileNotFoundError: [Errno 2] No such file or directory: '/content/2. LavadoActivos.xlsx'

2. Implementamos modelo Naive buyes.

In [ ]:
np.set_printoptions(suppress=True,precision=3)
mnb=GaussianNB()
mnb.fit(XD,yd)

#Mostramos las medias
u=mnb.theta_
sigma=mnb.var_;sigma=np.sqrt(sigma)
print("edad","ingresos_usd", "gastos_usd", "num_tarjetas_credito","monto_transado_tarjetas_usd","porcentaje_crecimiento_patrimonio", "lavado_activos")
print ("las medias son:")
print(u)
print ("las desviaciones son:")
print(sigma)
print('Los valores maximos son:\n',u+sigma)
print('Los valores minimos son:\n',u-sigma)

edad ingresos_usd gastos_usd num_tarjetas_credito monto_transado_tarjetas_usd porcentaje_crecimiento_patrimonio lavado_activos
las medias son:
[[   45.47   6690.733  3601.75      1.955  3090.464    44.687]
 [   44.3   70262.304 38707.284     2.016 33835.62     44.821]]
las desviaciones son:
[[    14.711   6206.964   3219.016      2.535   2456.859     14.173]
 [    15.052 113412.265  65902.557      2.557  62991.391     15.479]]
Los valores maximos son:
 [[    60.18   12897.697   6820.765      4.49    5547.323     58.86 ]
 [    59.352 183674.569 104609.842      4.574  96827.011     60.299]]
Los valores minimos son:
 [[    30.759    483.768    382.734     -0.581    633.605     30.514]
 [    29.248 -43149.96  -27195.273     -0.541 -29155.771     29.342]]


/usr/local/lib/python3.12/dist-packages/sklearn/utils/validation.py:1408: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


**Analisis de Resultado**

De los resultados arrojados por el modelo, se puede observar que la categoría de individuos que SÍ realizan lavado de activos (clase 1) presenta niveles significativamente más altos de ingresos, gastos y monto transado con tarjetas en comparación con la categoría de individuos que NO realizan lavado de activos (clase 0), lo cual evidencia la predominancia de las variables monetarias en la clasificación del modelo, ya que el lavado de activos se asocia principalmente a mayor volumen de movimiento financiero; es importante resaltar que aunque los ingresos y gastos promedio son considerablemente más altos en la clase 1, esta categoría presenta una alta dispersión en los valores, lo que indica la existencia de perfiles financieros muy diferentes dentro del grupo que sí realiza lavado, mientras que la clase 0 muestra un comportamiento más estable y homogéneo; por otro lado, la edad y el número de tarjetas de crédito presentan valores promedio muy similares entre ambas categorías, lo que sugiere que estas variables tienen un impacto limitado en la diferenciación entre individuos que realizan y no realizan lavado de activos, funcionando únicamente como variables de contexto; finalmente, el porcentaje de crecimiento patrimonial no evidencia diferencias relevantes entre las dos clases, lo que indica que, para este conjunto de datos, el modelo se apoya principalmente en la magnitud de los ingresos, gastos y transacciones como criterios determinantes para identificar posibles casos de lavado de activos, alineándose con los principios del análisis SARLAFT orientados a la detección de comportamientos financieros atípicos.

3. Se procede con la evaluación del modelo. Para la evaluación utilizamos la matriz de confusión.

In [ ]:
ydp=mnb.predict(XD) #Esto e lo que el modelo aprende
cm=confusion_matrix(yd,ydp)
print(cm)

#Se determinan las metricas de la matriz de confusion
VN=cm[0,0]; FP=cm[0,1]; FN=cm[1,0]; VP=cm[1,1]; TDatos=len(XBD)

#1. Exactitud: Funcionamiento general del modelo
Ex=(VP+VN)/TDatos
print("Exactitud=",Ex)

#2. Tasa Error: %Fallos del modelo
TE=(FP+FN)/TDatos
print("Tasa Error=",TE)

#3.Sensibilidad: Como se comparto con respectos a los Verdaderos Posotovps
Se=VP/(VP+FN)
print("Sensibilidad:", Se)

#4.Especificidad: como se comporta el pronostico negativos
Es=VN/(VN+FP)
print("Especificidad:", Es)

#5.Precision: Es una version de como se comporta el modelo frente a los positivos solamente

Pre=VP/(VP+FP)
print("Precision:", Pre)

#6.Prediccion Negativa: Como funciona el modelo pronosticando creditos pre negados

PreN=VN/(VN+FN)
print("Prediccion Negativa:", PreN)

[[2179   44]
 [ 145  782]]
Exactitud= 0.94
Tasa Error= 0.06
Sensibilidad: 0.8435814455231931
Especificidad: 0.9802069275753487
Precision: 0.9467312348668281
Prediccion Negativa: 0.9376075731497419


**Analisis del Resultado**

De acuerdo con los resultados obtenidos, se puede observar de manera general que el modelo, presenta un buen desempeño en la clasificación de individuos que realizan y no realizan lavado de activos, alcanzando una exactitud del 94%, lo que indica un comportamiento sólido del modelo frente a la detección de perfiles financieros sospechosos; se destaca especialmente la precisión, la cual alcanza un 94.67%, evidenciando que cuando el modelo clasifica a un individuo como sí realizador de lavado de activos, la probabilidad de que esta clasificación sea correcta es alta, lo que resulta fundamental en un contexto SARLAFT para evitar alertas innecesarias; igualmente, se observa un buen comportamiento del modelo frente a la identificación de individuos que no realizan lavado de activos, lo cual se refleja en una especificidad del 98.02%, indicando que el modelo es altamente efectivo al reconocer correctamente los casos no sospechosos; sin embargo, es importante mencionar que la sensibilidad, con un valor del 84.35%, muestra que aún existe un porcentaje de casos de lavado que no son detectados por el modelo, lo que impacta directamente la tasa de error del 6% y evidencia la necesidad de complementar el modelo con análisis adicionales, ya que en escenarios de lavado de activos los falsos negativos representan un riesgo significativo; finalmente, la predicción negativa del 93.76% confirma que el modelo mantiene un buen equilibrio en la clasificación de perfiles no sospechosos, consolidando al Naive Bayes esta herramienta como adecuada para el apoyo de la detección inicial de riesgos, aunque no como un mecanismo único de decisión.

4. Evaluacion de Lavado de Activos.

In [ ]:
XP= [34,1500,800,1000, 5000, 30] # Added two placeholder values to match model's expected 6 features
ydc=mnb.predict([XP])
print(ydc)

if ydc[0] == 1:
    print("El resultado es: Lavado de Activos") # Corrected indentation and made output more descriptive
else:
    print("El resultado es: No Lavado de Activos") # Corrected indentation and made output more descriptive

[1]
El resultado es: Lavado de Activos


/usr/local/lib/python3.12/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(
